$OpenBSD: patch-configure_ac,v 1.14 2017/05/08 10:46:39 ajacoutot Exp $

1. Old cdparanoia-3.a9.8 (cdparanoia-III-9.8) does not have cache control.

2. Avoid picking up Qt5 (used for examples only), or this will create
   a circular dependency.

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -319,6 +319,7 @@ PKG_CHECK_MODULES(QT, Qt5Gui Qt5Widgets, [
 ], [
   HAVE_QT=no
 ])
+HAVE_QT=no
 if test "x$HAVE_QT" = "xyes"; then
   AC_CHECK_TOOL(QT_MOC, moc)
   AC_MSG_CHECKING([for Qt5 moc])
@@ -630,6 +631,21 @@ AG_GST_CHECK_FEATURE(ALSA, [ALSA], alsa, [
   ])
 ])
 
+dnl *** sndio ***
+translit(dnm, m, l) AM_CONDITIONAL(USE_SNDIO, true)
+AG_GST_CHECK_FEATURE(SNDIO, [sndio audio], sndio, [
+  AC_CHECK_HEADER(sndio.h, HAVE_SNDIO="yes", HAVE_SNDIO="no")
+  if test $HAVE_SNDIO = yes
+  then
+    AC_CHECK_LIB(sndio, sio_open, HAVE_SNDIO="yes", HAVE_SNDIO="no", [])
+    if test $HAVE_SNDIO = yes
+    then
+      SNDIO_LIBS=-lsndio 
+      AC_SUBST(SNDIO_LIBS)
+    fi
+  fi
+])
+
 dnl *** CDParanoia ***
 dnl Note: upstream has a pkg-config file only in post-10.2 SVN so far
 translit(dnm, m, l) AM_CONDITIONAL(USE_CDPARANOIA, true)
@@ -660,10 +676,10 @@ AG_GST_CHECK_FEATURE(CDPARANOIA, [CDParanoia], cdparan
     if test "x$FOUND_CDPARANOIA" = "xyes";
     then
       AC_CHECK_LIB(cdda_paranoia,
-                   paranoia_cachemodel_size,
+                   paranoia_init,
                    : ,
                    HAVE_CDPARANOIA=no,
-  		 [-lcdda_interface])
+  		 [-lcdda_interface -lm])
     fi
 
     AC_SUBST(CDPARANOIA_LIBS)
@@ -774,6 +790,7 @@ else
 dnl not building plugins with external dependencies,
 dnl but we still need to set the conditionals
 AM_CONDITIONAL(USE_ALSA, false)
+AM_CONDITIONAL(USE_SNDIO, false)
 AM_CONDITIONAL(USE_CDPARANOIA, false)
 AM_CONDITIONAL(USE_IVORBIS, false)
 AM_CONDITIONAL(USE_LIBVISUAL, false)
@@ -895,6 +912,7 @@ sys/ximage/Makefile
 sys/xvimage/Makefile
 ext/Makefile
 ext/alsa/Makefile
+ext/sndio/Makefile
 ext/cdparanoia/Makefile
 ext/libvisual/Makefile
 ext/ogg/Makefile
